vendor/github.com/prometheus/client_golang/prometheus/summary.go

```diff
@@ -123,14 +115,8 @@ type SummaryOpts struct {
 	// Objectives defines the quantile rank estimates with their respective
 	// absolute error. If Objectives[q] = e, then the value reported for q
 	// will be the φ-quantile value for some φ between q-e and q+e.  The
-	// default value is DefObjectives. It is used if Objectives is left at
-	// its zero value (i.e. nil). To create a Summary without Objectives,
-	// set it to an empty map (i.e. map[float64]float64{}).
-	//
-	// Note that the current value of DefObjectives is deprecated. It will
-	// be replaced by an empty map in v1.0.0 of the library. Please
-	// explicitly set Objectives to the desired value to avoid problems
-	// during the transition.
```

Comment:
Are we bumping prometheus?

Reply:
Since we own the wrapping layer (i.e. component-base/metrics), we can actually just fallback to the old behavior by defaulting Objectives to whatever DefObjectives was.

Reply:
lint-dependencies.sh prompts me to upgrade github.com/prometheus/client_golang
```
    pinned:    v0.9.4
    preferred: v1.0.0
    hack/pin-dependency.sh github.com/prometheus/client_golang v1.0.
```

What shall I do?

Reply:
I think the bump of the dependency is fine (would like an ack from @logicalhan), but it introduces a breaking change we need to adjust for. The simplest thing we can do is In our SummaryOpts#toPromSummaryOpts method, we can preserve the existing behavior:

```go
if opts.Objectives == nil {
  // preserve compatibility with pre-1.0 prometheus usage
  opts.Objectives = map[float64]float64{0.5: 0.05, 0.9: 0.01, 0.99: 0.001}
}
```

Reply:
I have added @logicalhan 's commit for this https://github.com/kubernetes/kubernetes/pull/83987/commits/454a339f3afb9834c52ff5cace40ee6305a75d0c